.github/workflows/test.yml

```diff
@@ -0,0 +1,113 @@
+name: go
+on: [push]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.13
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+        with:
+          submodules: recursive
+
+      - name: Caching modules
+        uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-aw-${{ hashFiles('**/go.sum') }}
+
+      - name: Install dependency packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+          sudo apt-get install -y jq mesa-opencl-icd ocl-icd-opencl-dev pkg-config
+          curl https://sh.rustup.rs -sSf | sh -s -- -y
+          source $HOME/.cargo/env
+
+      - name: Get dependencies
+        run: |
+            export PATH=$PATH:$(go env GOPATH)/bin
+            source $HOME/.cargo/env
+            go get -u github.com/onsi/ginkgo/ginkgo
+            go get -u github.com/onsi/gomega/...
+            go get -u golang.org/x/lint/golint
+            go get -u github.com/loongy/covermerge
+            go get -u github.com/mattn/goveralls
+
+      - name: Run vetting
+        run: |
+            source $HOME/.cargo/env
+            cd $GITHUB_WORKSPACE/chain/filecoin/filecoin-ffi
+            make
+            cd $GITHUB_WORKSPACE
+            export PATH=$PATH:$(go env GOPATH)/bin
+            go vet ./...
+
+      - name: Run linting
+        run: |
+            cd $GITHUB_WORKSPACE
+            export PATH=$PATH:$(go env GOPATH)/bin
+            go get -u golang.org/x/lint/golint
+            golint ./...
+
+      - name: Run multichain infrastructure (bitcoin)
+        run: |
+            source $GITHUB_WORKSPACE/infra/.env
+            docker run -d -p 18443:18443 -h 0.0.0.0      \
+              --env-file $GITHUB_WORKSPACE/infra/.ci-env \
+              rohitnarurkar/multichain_bitcoin $BITCOIN_ADDRESS
```

Comment:
What is the reason for using `docker` and custom images instead of `docker-compose` and only specifying the chain being used at that moment? This would help test the `infra/` image definitions too.

Reply:
Yup, definitely something helpful!

> What is the reason for using docker and custom images instead of docker-compose

Nothing specific, at that time I didn't want the image build to consume time. But now that I am coming back to this, it seems building it also tests it! I'll add this